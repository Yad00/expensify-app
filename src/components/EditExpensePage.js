import React from "react";
import { useParams } from "react-router-dom";
import { connect } from "react-redux";
import { useNavigate  } from "react-router-dom";
import ExpenseForm from "./ExpenseForm";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

const EditExpensePage = (props) => {
    const { id } = useParams();
    const navigate = useNavigate();

    const expense = props.expenses.find((expense) => expense.id === id);
    
    return (
        <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Edit Expense</h1>
                </div>
            </div>
            <div className="content-container">
                <ExpenseForm
                    expense={expense}
                    onSubmit={(expense) => {
                        props.dispatch(startEditExpense(id, expense));
                        navigate('/');
                    }}
                />
                <button className="button button--secondary" onClick={() => {
                    props.dispatch(startRemoveExpense({ id }));
                    navigate('/');
                }}>Remove Expense</button>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        ...state
    };
};

export default connect(mapStateToProps)(EditExpensePage);