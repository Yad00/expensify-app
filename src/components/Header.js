import React from "react";
import { NavLink, useNavigate } from 'react-router-dom';
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";

export const Header = ({ startLogout }) => {
    const navigate = useNavigate();
    return (
        <header>
            <h1>Expensify</h1>
            <NavLink to="/dashboard" className={({ isActive }) => "nav-link" + (isActive ? " activated" : "")}>Dashboard</NavLink>
            <NavLink to="/create" className={({ isActive }) => "nav-link" + (isActive ? " activated" : "")}>Create Expense</NavLink>
            <button onClick={() => {
                startLogout();
                navigate('/');
            }}>Logout</button>
        </header>
)};

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);