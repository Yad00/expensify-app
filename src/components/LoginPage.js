import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";
import { useLocation, useNavigate } from 'react-router-dom';
import { auth } from '../firebase/firebase';

export const LoginPage = ({ startLogin }) => {
    let location = useLocation();
    const navigate = useNavigate();
    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__title">Expensify</h1>
                <p>Its time to get your expenses under control.</p>
                <button className="button" onClick={() => {
                    startLogin();
                    auth.onAuthStateChanged((user) => {
                        if (user) {
                            if (location.pathname === '/') {
                                navigate('/dashboard');
                            }
                        }
                    });
                }}>Login with Google</button>
            </div>
        </div>
    )
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);