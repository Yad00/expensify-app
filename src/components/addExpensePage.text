import React from "react";
import { connect } from "react-redux";
import { useNavigate  } from "react-router-dom";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        const navigate = useNavigate();
        this.props.onSubmit(expense);
        navigate('/');
    };
    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    onSubmit: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);