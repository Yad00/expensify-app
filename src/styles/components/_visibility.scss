.show-for-mobile {
    @media (min-width: $desktop-breakpoint) {
        display: none;
    }
}

.show-for-desktop {
    @media (max-width: $desktop-breakpoint-two) {
        display: none;
    }
}