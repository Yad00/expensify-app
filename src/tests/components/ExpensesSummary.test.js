import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpensesSummary";

test('Should correctly render expenses summary with one expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235}/>)
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expenses summary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={23} expensesTotal={234234213}/>)
    expect(wrapper).toMatchSnapshot();
});