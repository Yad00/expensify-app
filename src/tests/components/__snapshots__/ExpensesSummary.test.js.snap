// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $2,342,342.13
  </h1>
</div>
`;

exports[`Should correctly render expenses summary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;
